File: frontend/src/types/Movie.ts
export interface Movie {
  id: number;
  title: string;
  overview?: string;
  release_date?: string;
  vote_average?: number;
  vote_count?: number;
  status?: string;
  revenue?: number;
  runtime?: number;
  adult?: boolean;
  backdrop_path?: string;
  genres?: string;
  original_language?: string;
  poster_path?: string;
}

File: backend/app/models/movie.py
from typing import List
from pydantic import BaseModel
from typing import Optional

class Movie(BaseModel):
    id: int
    title: str
    overview: Optional[str] = None
    release_date: Optional[str] = None
    vote_average: Optional[float] = None
    vote_count: Optional[int] = None
    status: Optional[str] = None
    revenue: Optional[int] = None
    runtime: Optional[int] = None
    adult: Optional[bool] = None
    backdrop_path: Optional[str] = None
    genres: Optional[str] = None  # Se quiser, depois pode parsear como lista
    original_language: Optional[str] = None
    poster_path: Optional[str] = None
    
class MovieListResponse(BaseModel):
    items: List[Movie]
    total: int